docs/introduction/concepts.md

```diff
@@ -0,0 +1,281 @@
+
+Consider the following examples of rejections: 
+* `CreditCardValidationDeclined`, 
+* `OrderCannotBeEmpty`, 
+* `InsufficientFunds`.
+
+In Spine, rejections are defined as Protobuf messages in the file which names ends with
+`rejections.proto`. 
+
+For detailed instructions on defining rejections, please refer to
+[“Working with Rejections”](/docs/guides/rejections.html) guide.
+
+### Acknowledgement
+
+Acknowledgement is an outcome of sending a [Command](#command) to the [Command Service](#command-service).
+
+It tells whether the Command has been accepted for handling. 
+
+### Command Handler
+
+Command Handler is an object which receives commands, modifies the state of the application, and
+generates events if the modification was successful.
+
+[`Aggregate`](#aggregate) and [`ProcessManager`](#process-manager) are examples one of such classes. 
+The code snippet below given an example of handling a command by an aggregate:
+
+```
+final class TaskAggregate
+    extends Aggregate<TaskId, Task, Task.Builder> {
+    ...
+    @Assign
+    TaskCreated handle(CreateTask cmd, CommandContext ctx) {
+        return TaskCreated
+                .newBuilder()
+                .setId(cmd.getId())
+                .setName(cmd.getName())
+                .setOwner(ctx.getActor())
+                .vBuild();
+    }
+    ...
+}
+```
+
+### Event Subscriber
+
+Event Subscriber is an object which subscribes to receive events.
+
+The example below shows how a [Projection](#projection) class subscribed to the `TaskCompleted`
+event.
+
+  ```
+  final class TaskProjection
+      extends Projection<TaskId, TaskItem, TaskItem.Builder> {
+      ...
+      @Subscribe
+      void on(TaskCompleted e, EventContext ctx) {
+          builder().setWhenDone(ctx.getTimestamp());
+      }
+  }
+  ```
+
+### Event Reactor
+
+Event Reactor is an object which usually produces one or more events in response to an incoming
+event. Unlike [Event Subscriber](#event-subscriber), which always consumes events, a reacting object
+generates events in response to changes in the domain.
+
+<p class="note">In some cases, Event Reactor may ignore the event, returning `Nothing`.
+    It usually happens when a method returns one of the
+    [`Either`](https://spine.io/core-java/javadoc/server/index.html) types, with `Nothing` as
+    one of the possible options: `EitherOf2<TaskReAssigned, Nothing>`.</p>
+
+## Value Objects
+
+Value Object describe things in a domain model and do not have identity. 
+Value Objects are also immutable. Some examples are:
+  * `PhoneNumber`
+  * `EmailAddress`
+  * `BarCode`
+  
+In Spine, Value Objects are defined as Protobuf messages.
+
+## Entities
+
+Entities are main building blocks of a domain model. They have unique identity and modify their 
+state during the lifecycle.
+
+### Identifier
+ 
+The framework supports the following types of identifiers:
+
+* `Integer`,
+* `Long`,
+* `String`,
+* A generated Java class implementing the `Message`.
```

Comment:
Maybe we should put this first?

Reply:
I put it as the last item because:
  * I hope that the rule “The last is remembered” would work in this case.
  * The text that follows is based on the last item. Putting it first would break the logic flow.